src/style/design-tokens/generate-css-variables.util.spec.js

```diff
@@ -0,0 +1,42 @@
+import generateCssVariables from "./generate-css-variables.util";
+import baseTheme from "../themes/base";
+
+describe("generateCssVariables", () => {
+  const tokens = {
+    colorsActionMajor500: "pink",
+    colorsUtility500: "yellow",
+    colorsAction700: "black",
+  };
+  let theme;
+  let output;
+
+  beforeEach(() => {
+    theme = {
+      ...baseTheme,
+      ...tokens,
+    };
+    output = generateCssVariables(theme);
+  });
+
+  it("should generate multiline string", () => {
+    const linesNumber = output.split(/\r?\n/).length;
+
+    expect(typeof output).toEqual("string");
+    expect(linesNumber).toBeGreaterThan(0);
+  });
+
+  it("should generate css variables format for all tokens", () => {
+    const matches = output.match(/^--[a-zA-Z0-9]*:\s?.*;$/gm);
+
+    expect(Object.keys(tokens).length).toEqual(matches.length);
+  });
+
+  it("should remove baseTheme properties", () => {
+    const baseThemeProperties = Object.keys(baseTheme).join("|");
+    const baseThemePropertiesRegexp = new RegExp(
+      `^--(${baseThemeProperties}):.*\\$/gm`
+    );
+
+    expect(output).not.toMatch(baseThemePropertiesRegexp);
```

Comment:
I think it would be acceptable, and beneficial, to use a snapshot test here. It's difficult for maintainers to identify if the test is passing or the regex is incorrect. The snapshot would also have the advantage of showing when additional properties are added. We should snapshot each of the themes.